inc/class-post-type.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @package WPSEO\Inc
+ */
+
+/**
+ * Represents the post type utils.
+ */
+class WPSEO_Post_Type {
+
+	/**
+	 * Returns an array with the accessible post types.
+	 *
+	 * An accessible post type is a post type that is public and isn't set as no-index (robots)
+	 *
+	 * @return array
+	 */
+	public static function get_accessible_post_types() {
+		$post_types = get_post_types( array( 'public' => true ) );
+		$post_types = array_filter( $post_types, array( 'self', 'is_post_type_indexable' ) );
+
+		/**
+		 * Filter: 'wpseo_accessible_post_types' - Allow changing the accessible post types.
+		 *
+		 * @api array $post_types The public post types.
+		 */
+		$post_types = apply_filters( 'wpseo_accessible_post_types', $post_types );
+
+		// When the array gets messed up somewhere.
```

Comment:
This is inconsistent with the previous array checks where type-casting was being used. Please make this more uniform.